### PR TITLE
Add Fee History and MaxFeePerGas Endpoints [1.0.0-alpha8]

### PR DESCRIPTION
**Pull request by hayesgm**
This patch adds fee history and max fee per gas RPC endpoints and incorporates them when deciding EIP-1559 transaction gas.

### CODE REVIEW

**Review by fluffywaffles — APPROVED** [review]
:chefkiss:

File: lib/signet/fee_history.ex [hunk 1]
```diff
@@ -0,0 +1,99 @@
+defmodule Signet.FeeHistory do
+  @moduledoc ~S"""
+  Represents fee history data as defined in EIP-1559.
+
+  See `Signet.RPC.fee_history` for getting traces from
+  an Ethereum JSON-RPC host.
+
+  See also:
+    * Alcemy docs: https://docs.alchemy.com/reference/eth-feehistory
```
Comment (kevincheng96) [inline lib/signet/fee_history.ex:9]:
```suggestion
    * Alchemy docs: https://docs.alchemy.com/reference/eth-feehistory
```

**Review by fluffywaffles — APPROVED** [review]
looks great! now we just need to make sure our txs get mined... :pick: 

passing in buffer is good, let the client choose what heuristic (if any) to apply there with a reasonable default.

**Review by kevincheng96 — APPROVED** [review]
Looks great!

File: lib/signet/fee_history.ex (hunk 1 above)
Comment (hayesgm) [reply lib/signet/fee_history.ex]: fixed, thanks

**Pull request merged**